mitodl/mitxpro
# fix: Add content_type to revision.content for blog and webinar index pages

#### Pre-Flight checklist


#### What are the relevant tickets?
(Required)
Fix for https://github.com/mitodl/mitxpro/pull/2830

#### What's this PR do?
(Required)
Webinar and Blog index pages were created by the data migrations and with them, a revision was added with `content_type_id` in the `content_json` field.

Now wagtail has altered the type of `content_json` from Text to JSON field and also renamed it to `content`. With that wagtail has renamed `PageRevision` to `Revision` and added a new field `content_type` in it. Wagtail populates `Revision.content_type` using the `Revision.content.content_type` but for Blog and Webinar index page revisions, it does not exist. 

This results in null values for the `Revision.content_type` and when wagtail tries to add a constraint of `null=False`, migrations fail.

#### How should this be manually tested?

- Testing Broken State (As RC is Broken):
  - Remove existing containers
  - Checkout the `release` branch(Release does not contain the upgrade changes), run migrations, and create revisions for Blog and Webinar without content_type. Blog revision is created in migrations, webinar will need to be created for testing from the Django shell. Also, populate `content_type` for Catalog and Certificate revisions as we won't be fixing them in the migration and migrations would break because of them.
  - Checkout `master`, Run migrations, Migrations would fail at wagtailcore.0072
  - Checkout this branch, and run migrations. **Should succeed**.

- Test for Prod state
  - Checkout the `release` branch, run migrations, and create revisions for Blog and Webinar without content_type. Blog is created in migrations, webinar will need to be created for testing from the Django shell. Also, add content_type for Catalog and Certificate revisions as we won't be fixing them in the migration and migrations would break because of them.
  - Checkout this branch and run migrations. **Should Succeed.**
- Testing for any new instance in the Future.
  - Checkout this branch and run migrations. Should Succeed.
